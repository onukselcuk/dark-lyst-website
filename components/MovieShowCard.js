import { Fragment } from "react";
import HeartIcon from "./icons/HeartIcon";
import Link from "next/link";
import CircularRating from "./CircularRating";

const MovieShowCard = ({ cur, isHero, isShow, isProfile }) => {
	// const url = `https://image.tmdb.org/t/p/${isHero ? "w400" : "w300"}${cur.poster_path}`;
	const url = `https://cdn.onukselcuk.com/${isHero ? "w342" : "w300"}${cur.poster_path}`;
	let title = cur.name || cur.title;
	if (title.length > 25) {
		title = `${title.slice(0, 25)}...`;
	}
	let link;
	if (isShow) {
		link = `/show/detail/[tid]`;
	} else if (!isShow) {
		link = `/movie/detail/[pid]`;
	}
	let asLink;
	if (isShow) {
		asLink = `/show/detail/${cur.id}`;
	} else if (!isShow) {
		asLink = `/movie/detail/${cur.id}`;
	}

	const handleHeart = (e) => {
		e.preventDefault();
		alert("clicked");
	};

	let date;

	if (isProfile) {
		date = cur.release_date || cur.first_air_date;
		date = date ? date.slice(0, 4) : "";
	}

	return (
		<Link key={url} href={link} as={asLink}>
			<a className="container-link">
				<div className="tv-show-container">
					<div className="top-backdrop" />
					<img className="show-poster-image" src={url} alt={`${cur.name || cur.title} Poster Image`} />
					<div className="bottom-backdrop" />
					<div className="bottom-info-container">
						<div className="rating-container">
							<CircularRating rating={cur.vote_average} />
						</div>
						<div className="bottom-detail-container">
							<span className="show-name">{title}</span>
							{isProfile && (
								<Fragment>
									<p>
										<span>{date}</span> <span>{isShow ? "TV Show" : "Movie"}</span>
									</p>
									{isShow ? (
										<Fragment>
											<p>
												<span>{cur.job || cur.character}</span>
											</p>
											<p>
												<span>
													{cur.episode_count} Episode{cur.episode_count > 1 && "s"}
												</span>
											</p>
										</Fragment>
									) : (
										<p>
											<span>{cur.job || cur.character}</span>
										</p>
									)}
								</Fragment>
							)}
						</div>
					</div>

					<div onClick={handleHeart} className="heart-container">
						<HeartIcon />
					</div>
				</div>
				<style jsx>{`
					.container-link {
						text-decoration: none;
						color: #fff;
					}

					.container-link:active,
					.container-link:focus,
					.container-link:visited {
						text-decoration: none;
						color: #fff;
					}

					.tv-show-container {
						display: flex;
						flex-direction: column;
						align-items: center;
						position: relative;
						width: 95%;
						height: 100%;
						border-radius: 10px;
						overflow: hidden;
						margin: 0 auto;
					}

					.show-poster-image {
						width: 100%;
						height: 80%;
					}

					.top-backdrop {
						position: absolute;
						top: 0;
						left: 0;
						width: 100%;
						height: 35%;
						background-image: linear-gradient(to bottom, rgba(0, 0, 0, .7), transparent);
						opacity: 0;
						transition: all 400ms;
					}
					.tv-show-container:hover .top-backdrop {
						opacity: 1;
					}

					.bottom-backdrop {
						position: absolute;
						bottom: 0;
						left: 0;
						width: 100%;
						height: 35%;
						background-image: linear-gradient(to top, rgba(0, 0, 0, .7), transparent);
						opacity: 0;
						transition: all 400ms;
					}
					.tv-show-container:hover .bottom-backdrop {
						opacity: 1;
					}

					.bottom-info-container {
						width: 100%;
						height: 20%;
						background-color: rgba(0, 0, 0, .2);
						display: flex;
						align-items: center;
						padding: .5rem .5rem;
						z-index: 10;
					}

					.rating-container {
						width: 25%;
						height: 80%;
						z-index: 10;
						opacity: 1;
						margin: 0 1.5rem 0 1rem;
						transition: all 400ms;
						display: flex;
						align-items: center;
						${isHero && "width:60px; height:60px;"};
					}

					.bottom-detail-container {
						width: 75%;
					}

					.bottom-detail-container p {
						margin-bottom: 0;
					}

					.show-name {
						width: 70%;
						color: #fff;
						font-weight: bold;
						transition: all 400ms;
						font-size: 1.7rem;
						z-index: 10;
						${isHero && "font-size:2rem;"};
					}

					.heart-container {
						width: 35px;
						height: 35px;
						position: absolute;
						top: 2%;
						right: 7%;
						z-index: 10;
						opacity: 1;
						transition: all 400ms;
					}
				`}</style>
			</a>
		</Link>
	);
};

export default MovieShowCard;
