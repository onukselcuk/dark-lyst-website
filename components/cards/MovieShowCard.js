import { Fragment, useState, useEffect } from "react";
import HeartIcon from "../icons/HeartIcon";
import Link from "next/link";
import CircularRating from "../icons/CircularRating";
import { connect } from "react-redux";
import { toggleMovieHeart } from "../../store/actions/movieActions";
import { toggleShowHeart } from "../../store/actions/showActions";
import VisibilitySensor from "react-visibility-sensor";

const MovieShowCard = ({
    cur,
    isHero,
    isShow,
    isProfile,
    movieList,
    showList,
    toggleMovieHeart,
    toggleShowHeart,
    visibilityContainment
}) => {
    const url = `https://image.tmdb.org/t/p/${isHero ? "w342" : "w300"}${
        cur.poster_path
    }`;
    let title = cur.name || cur.title;
    if (title.length > 25) {
        title = `${title.slice(0, 25)}...`;
    }
    let link;
    if (isShow) {
        link = `/show/detail/[tid]`;
    } else if (!isShow) {
        link = `/movie/detail/[pid]`;
    }
    let asLink;
    if (isShow) {
        asLink = `/show/detail/${cur.id}`;
    } else if (!isShow) {
        asLink = `/movie/detail/${cur.id}`;
    }

    let isLiked;

    if (isShow) {
        isLiked = showList.some((current) => current.tmdbId === cur.id);
    } else if (!isShow) {
        isLiked = movieList.some((current) => current.tmdbId === cur.id);
    }

    const handleHeart = (e) => {
        e.preventDefault();
        if (isShow) {
            toggleShowHeart(cur, !isLiked);
        } else if (!isShow) {
            toggleMovieHeart(cur, !isLiked);
        }
    };

    let date;

    if (isProfile) {
        date = cur.release_date || cur.first_air_date;
        date = date ? date.slice(0, 4) : "";
    }

    const [isVisibleState, setIsVisibleState] = useState(false);

    const onVisibilityChange = (isVisible) => {
        if (isVisibleState !== true) {
            setIsVisibleState(isVisible);
        }
    };

    const [imageLoadedState, setImageLoadedState] = useState(false);

    const setImageLoaded = () => {
        setImageLoadedState(true);
    };

    const setImageLoadStart = () => {
        setImageLoadedState(false);
    };

    useEffect(() => {
        return () => {
            setIsVisibleState(false);
        };
    }, [cur]);

    return (
        <Link key={url} href={link} as={asLink}>
            <a className="container-link">
                <VisibilitySensor
                    onChange={onVisibilityChange}
                    partialVisibility={!isHero}
                    active={!isVisibleState}
                    containment={visibilityContainment}
                >
                    <div
                        className="tv-show-container"
                        style={{
                            opacity:
                                (isVisibleState || isHero) && imageLoadedState
                                    ? 1
                                    : 0,
                            transition: "opacity 400ms ease-in"
                        }}
                    >
                        <div className="top-backdrop" />

                        {isVisibleState && (
                            <img
                                className="show-poster-image"
                                src={url}
                                alt={`${cur.name || cur.title} Poster Image`}
                                onLoad={setImageLoaded}
                                onLoadStart={setImageLoadStart}
                            />
                        )}

                        <div className="bottom-backdrop" />
                        <div className="bottom-info-container">
                            <div className="rating-container">
                                <CircularRating rating={cur.vote_average} />
                            </div>
                            <div className="bottom-detail-container">
                                <span className="show-name">{title}</span>
                                {isProfile && (
                                    <Fragment>
                                        <p>
                                            <span>{date}</span>{" "}
                                            <span>
                                                {isShow ? "TV Show" : "Movie"}
                                            </span>
                                        </p>
                                        {isShow ? (
                                            <Fragment>
                                                <p>
                                                    <span>
                                                        {cur.job ||
                                                            cur.character}
                                                    </span>
                                                </p>
                                                <p>
                                                    <span>
                                                        {cur.episode_count}{" "}
                                                        Episode
                                                        {cur.episode_count >
                                                            1 && "s"}
                                                    </span>
                                                </p>
                                            </Fragment>
                                        ) : (
                                            <p>
                                                <span>
                                                    {cur.job || cur.character}
                                                </span>
                                            </p>
                                        )}
                                    </Fragment>
                                )}
                            </div>
                        </div>

                        <div onClick={handleHeart} className="heart-container">
                            <HeartIcon isLiked={isLiked} />
                        </div>
                    </div>
                </VisibilitySensor>
                <style jsx>{`
                    .container-link {
                        text-decoration: none;
                        color: #fff;
                    }

                    .container-link:active,
                    .container-link:focus,
                    .container-link:visited {
                        text-decoration: none;
                        color: #fff;
                    }

                    .tv-show-container {
                        display: flex;
                        flex-direction: column;
                        align-items: center;
                        position: relative;
                        width: 95%;
                        height: 100%;
                        border-radius: 10px;
                        overflow: hidden;
                        margin: 0 auto;
                    }

                    .show-poster-image {
                        width: 100%;
                        height: 80%;
                    }

                    .top-backdrop {
                        position: absolute;
                        top: 0;
                        left: 0;
                        width: 100%;
                        height: 35%;
                        background-image: linear-gradient(
                            to bottom,
                            rgba(0, 0, 0, 0.7),
                            transparent
                        );
                        opacity: 0;
                        transition: all 400ms;
                    }
                    .tv-show-container:hover .top-backdrop {
                        opacity: 1;
                    }

                    .bottom-backdrop {
                        position: absolute;
                        bottom: 0;
                        left: 0;
                        width: 100%;
                        height: 35%;
                        background-image: linear-gradient(
                            to top,
                            rgba(0, 0, 0, 0.7),
                            transparent
                        );
                        opacity: 0;
                        transition: all 400ms;
                    }
                    .tv-show-container:hover .bottom-backdrop {
                        opacity: 1;
                    }

                    .bottom-info-container {
                        width: 100%;
                        height: 20%;
                        background-color: rgba(0, 0, 0, 0.2);
                        display: flex;
                        align-items: center;
                        padding: 0.5rem 0.5rem;
                        z-index: 10;
                    }

                    .rating-container {
                        width: 25%;
                        height: 80%;
                        z-index: 10;
                        opacity: 1;
                        margin: 0 1.5rem 0 1rem;
                        transition: all 400ms;
                        display: flex;
                        align-items: center;
                        ${isHero && "width:60px; height:60px;"};
                    }

                    .bottom-detail-container {
                        width: 75%;
                    }

                    .bottom-detail-container p {
                        margin-bottom: 0;
                    }

                    .show-name {
                        width: 70%;
                        color: #fff;
                        font-weight: bold;
                        transition: all 400ms;
                        font-size: 1.7rem;
                        z-index: 10;
                        ${isHero && "font-size:2rem;"};
                    }

                    .heart-container {
                        width: 35px;
                        height: 35px;
                        position: absolute;
                        top: 2%;
                        right: 7%;
                        z-index: 10;
                        opacity: 1;
                        transition: all 400ms;
                    }

                    .show-poster-image {
                        width: 100%;
                    }
                `}</style>
            </a>
        </Link>
    );
};

const mapStateToProps = (state) => {
    return {
        movieList: state.movies.movieList,
        showList: state.shows.showList
    };
};

export { MovieShowCard };

export default connect(mapStateToProps, { toggleMovieHeart, toggleShowHeart })(
    MovieShowCard
);
