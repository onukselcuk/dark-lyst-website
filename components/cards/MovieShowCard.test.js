import { shallow } from "enzyme";
import { MovieShowCard } from "./MovieShowCard";

const propsMovie = {
    cur: {
        backdrop_path: "/o0F8xAt8YuEm5mEZviX5pEFC12y.jpg",
        genre_ids: [12, 14, 878, 10751],
        id: 475430,
        media_type: "movie",
        original_language: "en",
        original_title: "Artemis Fowl",
        popularity: 224.04,
        poster_path: "/mhDdx7o7hhrxrikq8aqPLLnS9w8.jpg",
        release_date: "2020-06-12",
        title: "Artemis Fowl",
        vote_average: 6,
        vote_count: 334
    },
    isHero: true,
    isProfile: undefined,
    isShow: undefined,
    movieList: [],
    showList: []
};

const propsShow = {
    cur: {
        backdrop_path: "/fc9ebsTmVCvv6jjD2qT7rWazaej.jpg",
        first_air_date: "2018-06-06",
        genre_ids: [18],
        id: 71146,
        name: "Condor",
        origin_country: ["US"],
        original_language: "en",
        original_name: "Condor",
        popularity: 41.434,
        poster_path: "/28cIK70tN2t4gPTV8CBQZED1H2G.jpg",
        vote_average: 6.5,
        vote_count: 72
    },
    isHero: undefined,
    isProfile: undefined,
    isShow: true,
    movieList: [],
    showList: []
};

describe("When MovieShowCard renders", () => {
    it("expect to render a movie", () => {
        const wrapper = shallow(<MovieShowCard {...propsMovie} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("expect to render a show", () => {
        const wrapper = shallow(<MovieShowCard {...propsShow} />);
        expect(wrapper).toMatchSnapshot();
    });
});
