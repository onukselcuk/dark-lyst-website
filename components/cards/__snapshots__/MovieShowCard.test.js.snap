// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When MovieShowCard renders expect to render a movie 1`] = `
<Link
  as="/movie/detail/475430"
  href="/movie/detail/[pid]"
  key="https://image.tmdb.org/t/p/w342/mhDdx7o7hhrxrikq8aqPLLnS9w8.jpg"
>
  <a
    className="jsx-1306811756 container-link"
  >
    <div
      className="jsx-1306811756 tv-show-container"
    >
      <div
        className="jsx-1306811756 top-backdrop"
      />
      <img
        alt="Artemis Fowl Poster Image"
        className="jsx-1306811756 show-poster-image"
        src="https://image.tmdb.org/t/p/w342/mhDdx7o7hhrxrikq8aqPLLnS9w8.jpg"
      />
      <div
        className="jsx-1306811756 bottom-backdrop"
      />
      <div
        className="jsx-1306811756 bottom-info-container"
      >
        <div
          className="jsx-1306811756 rating-container"
        >
          <CircularRating
            rating={6}
          />
        </div>
        <div
          className="jsx-1306811756 bottom-detail-container"
        >
          <span
            className="jsx-1306811756 show-name"
          >
            Artemis Fowl
          </span>
        </div>
      </div>
      <div
        className="jsx-1306811756 heart-container"
        onClick={[Function]}
      >
        <HeartIcon
          isLiked={false}
        />
      </div>
    </div>
    <JSXStyle
      dynamic={
        Array [
          "width:60px; height:60px;",
          "font-size:2rem;",
        ]
      }
      id="3225958809"
    >
      .container-link.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;color:#fff;}.container-link.__jsx-style-dynamic-selector:active,.container-link.__jsx-style-dynamic-selector:focus,.container-link.__jsx-style-dynamic-selector:visited{-webkit-text-decoration:none;text-decoration:none;color:#fff;}.tv-show-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;width:95%;height:100%;border-radius:10px;overflow:hidden;margin:0 auto;}.show-poster-image.__jsx-style-dynamic-selector{width:100%;height:80%;}.top-backdrop.__jsx-style-dynamic-selector{position:absolute;top:0;left:0;width:100%;height:35%;background-image:linear-gradient( to bottom, rgba(0,0,0,0.7), transparent );opacity:0;-webkit-transition:all 400ms;transition:all 400ms;}.tv-show-container.__jsx-style-dynamic-selector:hover .top-backdrop.__jsx-style-dynamic-selector{opacity:1;}.bottom-backdrop.__jsx-style-dynamic-selector{position:absolute;bottom:0;left:0;width:100%;height:35%;background-image:linear-gradient( to top, rgba(0,0,0,0.7), transparent );opacity:0;-webkit-transition:all 400ms;transition:all 400ms;}.tv-show-container.__jsx-style-dynamic-selector:hover .bottom-backdrop.__jsx-style-dynamic-selector{opacity:1;}.bottom-info-container.__jsx-style-dynamic-selector{width:100%;height:20%;background-color:rgba(0,0,0,0.2);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0.5rem 0.5rem;z-index:10;}.rating-container.__jsx-style-dynamic-selector{width:25%;height:80%;z-index:10;opacity:1;margin:0 1.5rem 0 1rem;-webkit-transition:all 400ms;transition:all 400ms;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:60px; height:60px;;}.bottom-detail-container.__jsx-style-dynamic-selector{width:75%;}.bottom-detail-container.__jsx-style-dynamic-selector p.__jsx-style-dynamic-selector{margin-bottom:0;}.show-name.__jsx-style-dynamic-selector{width:70%;color:#fff;font-weight:bold;-webkit-transition:all 400ms;transition:all 400ms;font-size:1.7rem;z-index:10;font-size:2rem;;}.heart-container.__jsx-style-dynamic-selector{width:35px;height:35px;position:absolute;top:2%;right:7%;z-index:10;opacity:1;-webkit-transition:all 400ms;transition:all 400ms;}
    </JSXStyle>
  </a>
</Link>
`;

exports[`When MovieShowCard renders expect to render a show 1`] = `
<Link
  as="/show/detail/71146"
  href="/show/detail/[tid]"
  key="https://image.tmdb.org/t/p/w300/28cIK70tN2t4gPTV8CBQZED1H2G.jpg"
>
  <a
    className="jsx-1742891495 container-link"
  >
    <div
      className="jsx-1742891495 tv-show-container"
    >
      <div
        className="jsx-1742891495 top-backdrop"
      />
      <img
        alt="Condor Poster Image"
        className="jsx-1742891495 show-poster-image"
        src="https://image.tmdb.org/t/p/w300/28cIK70tN2t4gPTV8CBQZED1H2G.jpg"
      />
      <div
        className="jsx-1742891495 bottom-backdrop"
      />
      <div
        className="jsx-1742891495 bottom-info-container"
      >
        <div
          className="jsx-1742891495 rating-container"
        >
          <CircularRating
            rating={6.5}
          />
        </div>
        <div
          className="jsx-1742891495 bottom-detail-container"
        >
          <span
            className="jsx-1742891495 show-name"
          >
            Condor
          </span>
        </div>
      </div>
      <div
        className="jsx-1742891495 heart-container"
        onClick={[Function]}
      >
        <HeartIcon
          isLiked={false}
        />
      </div>
    </div>
    <JSXStyle
      dynamic={
        Array [
          undefined,
          undefined,
        ]
      }
      id="3225958809"
    >
      .container-link.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;color:#fff;}.container-link.__jsx-style-dynamic-selector:active,.container-link.__jsx-style-dynamic-selector:focus,.container-link.__jsx-style-dynamic-selector:visited{-webkit-text-decoration:none;text-decoration:none;color:#fff;}.tv-show-container.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;width:95%;height:100%;border-radius:10px;overflow:hidden;margin:0 auto;}.show-poster-image.__jsx-style-dynamic-selector{width:100%;height:80%;}.top-backdrop.__jsx-style-dynamic-selector{position:absolute;top:0;left:0;width:100%;height:35%;background-image:linear-gradient( to bottom, rgba(0,0,0,0.7), transparent );opacity:0;-webkit-transition:all 400ms;transition:all 400ms;}.tv-show-container.__jsx-style-dynamic-selector:hover .top-backdrop.__jsx-style-dynamic-selector{opacity:1;}.bottom-backdrop.__jsx-style-dynamic-selector{position:absolute;bottom:0;left:0;width:100%;height:35%;background-image:linear-gradient( to top, rgba(0,0,0,0.7), transparent );opacity:0;-webkit-transition:all 400ms;transition:all 400ms;}.tv-show-container.__jsx-style-dynamic-selector:hover .bottom-backdrop.__jsx-style-dynamic-selector{opacity:1;}.bottom-info-container.__jsx-style-dynamic-selector{width:100%;height:20%;background-color:rgba(0,0,0,0.2);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0.5rem 0.5rem;z-index:10;}.rating-container.__jsx-style-dynamic-selector{width:25%;height:80%;z-index:10;opacity:1;margin:0 1.5rem 0 1rem;-webkit-transition:all 400ms;transition:all 400ms;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;undefined;}.bottom-detail-container.__jsx-style-dynamic-selector{width:75%;}.bottom-detail-container.__jsx-style-dynamic-selector p.__jsx-style-dynamic-selector{margin-bottom:0;}.show-name.__jsx-style-dynamic-selector{width:70%;color:#fff;font-weight:bold;-webkit-transition:all 400ms;transition:all 400ms;font-size:1.7rem;z-index:10;undefined;}.heart-container.__jsx-style-dynamic-selector{width:35px;height:35px;position:absolute;top:2%;right:7%;z-index:10;opacity:1;-webkit-transition:all 400ms;transition:all 400ms;}
    </JSXStyle>
  </a>
</Link>
`;
